### PR TITLE
Fix: Fixed issue where the option to leave compact overlay didn't work

### LINKED ISSUE #9928: Bug: Option to leave compact overlay fails
**Steps to repro**
- Open Files
- Enter compact overlay
- Click option to leave compact overlay and observe error

**Steps to fix**
- Leave compact overlay and go back to the regular view

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9928 

**Changes**
- Removed unused methods in `InnerNavigationToolbar.caml.cs` (`NavToolbarEnterCompactOverlay_Click()` and `SetAppBarButtonProps()`)
- Changed `Setter` of `IsWindowCompactOverlay` (`MainPageViewModel.cs`) as the existing one was not affecting the backing `DependencyProperty`
- Fixed logic-error in `SetCompactOverlay()` (`MainPage.xaml.cs`) line@ 508 and 513

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files.App/ViewModels/MainPageViewModel.cs [hunk 1]
```diff
@@ -80,10 +75,11 @@ public MainPageViewModel()
 			OpenSettingsCommand = new RelayCommand(OpenSettings);
 		}
 
-		private void NavigateToNumberedTabKeyboardAccelerator(KeyboardAcceleratorInvokedEventArgs e)
+		private void NavigateToNumberedTabKeyboardAccelerator(KeyboardAcceleratorInvokedEventArgs? e)
 		{
 			int indexToSelect = 0;
-
+			if (e is null)
```
Comment (yaichenbaum) [inline src/Files.App/ViewModels/MainPageViewModel.cs:81]: Is e ever null?

Comment (ferrariofilippo) [reply src/Files.App/ViewModels/MainPageViewModel.cs:81]: Just checked and it is never null.
Should I remove nullable annotations or should I use `!` to ignore them?

Comment (yaichenbaum) [reply src/Files.App/ViewModels/MainPageViewModel.cs:81]: 👍

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Pull request merged**